### PR TITLE
[Enhancement] Refactor SDK pipeline 

### PR DESCRIPTION
**Pull request by lzhangzz**
usage

`det_cls.cpp`
`det_pose.cpp`
`text_det_recog.cpp`

### CODE REVIEW

File: demo/python/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                output='*labels')
+        ],
+        output=['dets', 'labels'])
+
+    device = Device(args.device)
+    pipeline = Pipeline(config, Context(device))
+
+    img = cv2.imread(args.image_path)
+
+    output = pipeline(dict(ori_img=img))
```
Comment (irexyc) [inline demo/python/pipeline.py:61]: How to pass multi images?

File: csrc/mmdeploy/codebase/mmcls/linear_cls.cpp [hunk 2]
```diff
@@ -68,4 +69,34 @@ class LinearClsHead : public MMClassification {
 
 REGISTER_CODEBASE_COMPONENT(MMClassification, LinearClsHead);
 
+class CropBox {
```
Comment (lvhan028) [inline csrc/mmdeploy/codebase/mmcls/linear_cls.cpp:72]: Is there any specific reason to add `CropBox` in `mmcls`?

File: csrc/mmdeploy/core/graph.cpp [hunk 3]
```diff
@@ -4,26 +4,137 @@
...
+  return std::move(builder);
+}
+
+Result<std::vector<std::string>> ParseStringArray(const Value& value) {
+  if (value.is_string()) {
+    return std::vector{value.get<std::string>()};
+  } else if (value.is_array()) {
+    return from_value<std::vector<std::string>>(value);
+  }
+  MMDEPLOY_ERROR("{}", value);
```
Comment (lvhan028) [inline csrc/mmdeploy/core/graph.cpp:132]: May use `MMDEPLOY_DEBUG`

File: csrc/mmdeploy/graph/task.cpp [hunk 4]
```diff
@@ -39,48 +39,63 @@ Sender<Value> Task::Process(Sender<Value> input) {
   });
 }
 
-Result<unique_ptr<Task>> TaskParser::Parse(const Value& config) {
+TaskBuilder::TaskBuilder(Value config) : Builder(std::move(config)) {
+  // MMDEPLOY_CRITICAL("{}", config_);
```
Comment (lvhan028) [inline csrc/mmdeploy/graph/task.cpp:43]: May remove useless code

**Comment by lvhan028** [comment]
Got some questions: 
- If one classification task is followed by two detection tasks, what does the config look like?
- How to set the batch size?

**Review by lvhan028 — APPROVED** [review]

**Comment by lzhangzz** [comment]
> * How to set the batch size?

`NetModule` has nothing to do with batch size for now.

> * If one detection task is followed by two classification tasks, will the `CropBox` node be shared?

Yes

> * How to ensure that the output sequence is the same as the input sequence?

Ensured by the sender-receiver model

File: demo/python/pipeline.py (hunk 1 above)
Comment (lzhangzz) [reply demo/python/pipeline.py:61]: same as other tasks' batch inference API `pipeline.batch(...)`

**Review by irexyc — APPROVED** [review]

**Comment by zml24** [comment]
Seems this commit failed for Python SDK.
```python
import cv2
from mmdeploy_python import Detector
img = cv2.imread('mmdetection/demo/demo.jpg')
detector = Detector(model_path='mmdeploy_model/faster-rcnn/tensorrt', device_name='cuda', device_id=0)
bboxes, labels, _ = detector(img)
```
```shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: array has incorrect number of dimensions: 0; expected 2
```

**Pull request merged**